dsp/primitives/predict.py

```diff
@@ -111,6 +111,7 @@ def do_generate(
             max_tokens_key = "max_tokens" if "max_tokens" in keys else "max_output_tokens"
             new_kwargs = {
                 **kwargs,
```

Comment:
How the value set by the user effects the maximum tokens passed and returned from the model is quite unclear here, would anyone be able to clarify how 101:118 relate to the eventual values passed to different models, I'll have a crack at clearing this up. Might it be possible to do the `75` checking upstream of here?